src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -304,8 +304,10 @@ private void SolutionEvents_AfterClosing()
             }
         }
 
-        public void CleanCache()
+        public async Task CleanCacheAsync()
         {
+            await SolutionManager.CleanAsync();
```

Comment:
Because this `cleanAsync` goes through all the build integrated nuget projects and then get their cache file path and delete it. I'll rather have this code inside VSSolutionManager which is where we deal with NuGet projects then bring it into SolutionRestoreWorker. Worker doesn't deal with nuget projects and neither it needs to.